core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -27,7 +27,10 @@ object Log4jControllerRegistration {
   try {
     val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
     val instance = log4jController.getDeclaredConstructor().newInstance()
-    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+    val success = CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+    if (!success) {
+      throw new IllegalArgumentException
```

Comment:
This should have a message. Also, we should decide if we want to shutdown or if we want to log it. I usually prefer fail-fast, but not sure in this case. For example, what happens if there are multiple brokers running in the same jvm?